player/history_heuristic/history_heuristic_node.py

```diff
@@ -0,0 +1,36 @@
+import random
+
+from game_logic.game_status import GameStatus
+from game_logic.igame_state import IGameState
+from player.mcts.imcts_node import IMctsNode
+
+from player.mcts.mcts_node import MctsNode
+
+
+class HistoryHeuristicNode(MctsNode):
+
+    def __init__(self, game_state: IGameState, action: int, parent: IMctsNode, epsilon = 0):
+        super().__init__(game_state, action, parent)
+        self._epsilon = epsilon
+
+    @property
+    def epsilon(self) -> float:
+        return self._epsilon
+
+    def simulate(self, state):
+        actions_history = self.get_actions_history_dictionary()
+        while state.game_status == GameStatus.InProgress:
+            p = random.random()
+            if p < self.epsilon:
+                # Get first available move based on actions history
+                for action in actions_history:
```

Comment:
you can write it in one line
`state.make_move(next(action for action in actions_history if action in state.available_moves), default=-1)`
next will return -1 if there is no available moves.

Reply:
Yep, thanks for advice. Changed to next and set default to random available move.